virtcontainers/clh.go

```diff
@@ -0,0 +1,1260 @@
+// Copyright (c) 2019 Ericsson Eurolab Deutschland GmbH
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	persistapi "github.com/kata-containers/runtime/virtcontainers/persist/api"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/kata-containers/runtime/virtcontainers/device/config"
+	"github.com/kata-containers/runtime/virtcontainers/store"
+	"github.com/kata-containers/runtime/virtcontainers/types"
+	"github.com/kata-containers/runtime/virtcontainers/utils"
+)
+
+//
+// Constants and type definitions related to cloud hypervisor
+//
+
+type clhState uint8
+
+const (
+	clhNotReady clhState = iota
+	clhReady
+)
+
+const (
+	clhTimeout            = 10
+	clhSocket             = "clh.sock"
+	vfsdSocket            = "virtiofsd.sock"
+	clhSerial             = "serial-tty.log"
+	clhStopSandboxTimeout = 15
+	supportedMajorVersion = 0
+	supportedMinorVersion = 3
+	defaultIchPath        = "/usr/local/bin/cloud-hypervisor"
+)
+
+type CloudHypervisorVersion struct {
+	Major    int
+	Minor    int
+	Revision int
+}
+
+//
+// Cloud hypervisor state
+//
+type CloudHypervisorState struct {
+	state        clhState
+	PID          int
+	VirtiofsdPID int
+}
+
+type cloudHypervisor struct {
+	id         string
+	state      CloudHypervisorState
+	clhd       *exec.Cmd
+	store      *store.VCStore
+	config     HypervisorConfig
+	ctx        context.Context
+	socketPath string
+	version    CloudHypervisorVersion
+	cliBuilder *DefaultCLIBuilder
+}
+
+var clhKernelParams = []Param{
+
+	{"root", "/dev/vda1"},
+	{"panic", "1"},              // upon kernel panic wait 1 second before reboot
+	{"no_timer_check", ""},      // do not check broken timer IRQ resources
+	{"noreplace-smp", ""},       // do not replace SMP instructions
+	{"page_alloc.shuffle", "1"}, // enable page allocation shuffling
```

Comment:
Any reason why we'd want that one by default? Also, afair this is only available in fairly recent kernels. I think this should be left as a `kernel_params` configuration file addition.

Reply:
Ok I can remove it. It's a security feature indeed quite recent